### PR TITLE
Install all dependencies

### LINKED ISSUE #67: TOPP Tools in Windows executables
Still some issues with running TOPP tools in Windows executables.

- works in streamlit-template built with OpenMS release version 3.0.0
- does NOT work in umetaflow with OpenMS develop branch or release version 3.1.0
- what about other apps?
- Need to check path to correct TOPP tool in input_TOPP widget for ini file creation and running the TOPP tool

### PR DESCRIPTION
**Pull request by t0mdavid-m**
### **User description**
This PR adds portable versions of all dependencies to the windows installations using the `cpack` routine from the installer.

Note that this required some changes to the OpenMS build configuration and is thus dependent on the following PR in the main repository:

https://github.com/OpenMS/OpenMS/pull/7529

This should fix #67.

___

### **PR Type**
enhancement, configuration changes

___

### **Description**
- Added a new packaging step using `ctest` and `cpack` to both GitHub Actions workflows.
- Replaced separate artifact uploads for bin and share directories with a single package upload.
- Updated artifact download and extraction steps to handle the new package format.
- Removed steps for deleting individual bin and share artifacts, replaced with package artifact deletion.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>build-windows-executable-app-with-pyinstaller.yaml</strong><dd><code>Add packaging step and consolidate artifact handling</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
.github/workflows/build-windows-executable-app-with-pyinstaller.yaml

<li>Added a new packaging step using <code>ctest</code> and <code>cpack</code>.<br> <li> Replaced separate artifact uploads for bin and share directories with <br>a single package upload.<br> <li> Updated artifact download and extraction steps to handle the new <br>package format.<br> <li> Removed steps for deleting individual bin and share artifacts, <br>replaced with package artifact deletion.<br>

</details>
    

  </td>
  <td><a href="https://github.com/OpenMS/streamlit-template/pull/68/files#diff-02740fb40ac476b44a312979c501dcdd042c91d20cd21389ee0fb446f96bb07c">+28/-26</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>build-windows-executable-app.yaml</strong><dd><code>Add packaging step and consolidate artifact handling</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
.github/workflows/build-windows-executable-app.yaml

<li>Added a new packaging step using <code>ctest</code> and <code>cpack</code>.<br> <li> Replaced separate artifact uploads for bin and share directories with <br>a single package upload.<br> <li> Updated artifact download and extraction steps to handle the new <br>package format.<br> <li> Removed steps for deleting individual bin and share artifacts, <br>replaced with package artifact deletion.<br>

</details>
    

  </td>
  <td><a href="https://github.com/OpenMS/streamlit-template/pull/68/files#diff-211d58f377673fc6e00e49835e1cf5dc727c333368e2b5f8f1cf31e1eff077c5">+26/-24</a>&nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by t0mdavid-m** [comment]
The workflow is currently failing since `cipackage.cmake` was introduced after the 3.0.0 release. However, until the PR is merged and a release is created, there wont be an available release that supports the new workflow.

**Comment by t0mdavid-m** [comment]
I updated the delete artifact version (see #71)

**Comment by t0mdavid-m** [comment]
The PR (https://github.com/OpenMS/OpenMS/pull/7529) in the main repository is now merged. We decided on different flags which I just updated for this PR as well.

File: .github/workflows/build-windows-executable-app-with-pyinstaller.yaml [hunk 1]
```diff
@@ -216,14 +223,23 @@ jobs:
...
     - name: Copy everything to dist directory
       run: |
+        mkdir dist
+        mkdir dist/bin
+        
         cp -r .streamlit dist/.streamlit
         cp -r content dist/content
+        cp -r docs dist/docs
         cp -r src dist/src
         cp -r assets dist/assets
```
Comment (Arslan-Siraj) [inline .github/workflows/build-windows-executable-app-with-pyinstaller.yaml:233]: gdpr_consent needed here?

File: .github/workflows/build-windows-executable-app.yaml
Comment (Arslan-Siraj) [inline .github/workflows/build-windows-executable-app.yaml:1]: still bin folder which is empty.

File: .github/workflows/build-windows-executable-app-with-pyinstaller.yaml (hunk 1 above)
Comment (t0mdavid-m) [reply .github/workflows/build-windows-executable-app-with-pyinstaller.yaml]: Should be able to remove it!

**Comment by t0mdavid-m** [comment]
PR would be ready from my side! @Arslan-Siraj

**Pull request merged**